accio-main/src/main/java/io/accio/main/wireprotocol/PgMetastoreImpl.java

```diff
@@ -0,0 +1,155 @@
+{
+    private static final Logger LOG = Logger.get(PgMetastoreImpl.class);
+    private final ConfigManager configManager;
+    private final DuckDBSqlConverter duckDBSqlConverter;
+    private final DuckdbClient duckdbClient;
+
+    @Inject
+    public PgMetastoreImpl(
+            ConfigManager configManager,
+            DuckDBSqlConverter duckDBSqlConverter)
+    {
+        this.configManager = requireNonNull(configManager, "configManager is null");
+        this.duckDBSqlConverter = duckDBSqlConverter;
+        this.duckdbClient = buildDuckDBClient();
+    }
+
+    @Override
+    public boolean isSchemaExist(String name)
+    {
+        try (AutoCloseableIterator<Object[]> iter = duckdbClient
+                .query("SELECT 1 FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = ?", List.of(new Parameter(VarcharType.VARCHAR, name)))) {
+            return iter.hasNext();
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void dropTableIfExists(String name)
+    {
+        try {
+            duckdbClient.executeDDL(format("BEGIN TRANSACTION;DROP TABLE IF EXISTS %s;COMMIT;", name));
+        }
+        catch (Exception e) {
+            LOG.error(e, "Failed to drop table %s", name);
+        }
+    }
+
+    @Override
+    public void directDDL(String sql)
+    {
+        duckdbClient.executeDDL(sql);
+    }
+
+    @Override
+    public ConnectorRecordIterator directQuery(String sql, List<Parameter> parameters)
+    {
+        try {
+            return DuckdbRecordIterator.of(duckdbClient, sql, convertParameters(parameters));
+        }
+        catch (Exception e) {
+            throw new AccioException(GENERIC_INTERNAL_ERROR, e);
+        }
+    }
+
+    @Override
+    public String getPgCatalogName()
+    {
+        return PG_CATALOG_NAME;
+    }
+
+    @Override
+    public String handlePgType(String type)
+    {
+        if (type.startsWith("_")) {
+            return format("%s[]", handlePgType(type.substring(1)));
+        }
+        else if (!DuckdbTypes.getDuckDBTypeNames().contains(type)) {
+            return "VARCHAR";
+        }
+        return type;
+    }
+
+    @Override
+    public Client getClient()
+    {
+        return duckdbClient;
+    }
+
+    @Override
+    public SqlConverter getSqlConverter()
+    {
+        return duckDBSqlConverter;
+    }
+
+    @Override
+    public void close()
+    {
+        duckdbClient.close();
+    }
+
+    private DuckdbClient buildDuckDBClient()
+    {
+        return DuckdbClient.builder()
+                .setDuckDBConfig(configManager.getConfig(DuckDBConfig.class))
+                .setCacheStorageConfig(getCacheStorageConfigIfExists())
+                .build();
+    }
+
+    private CacheStorageConfig getCacheStorageConfigIfExists()
+    {
+        try {
+            return configManager.getConfig(CacheStorageConfig.class);
+        }
+        catch (Exception e) {
+            return null;
```

Comment:
```suggestion
        catch (Exception e) {
            LOG.warn(e, "%s connector does not support cache storage. Cache is disable.", configManager.getConfig(AccioConfig.class).getDataSourceType().name());
            return null;
```
Let's add some log message.